actiontext/lib/action_text/engine.rb

```diff
@@ -37,21 +37,24 @@ def to_trix_content_attachment_partial_path
     end
 
     initializer "action_text.helper" do
-      ActiveSupport.on_load(:action_controller_base) do
-        helper ActionText::Engine.helpers
+      %i[action_controller_base action_mailer].each do |abstract_controller|
+        ActiveSupport.on_load(abstract_controller) do
+          helper ActionText::Engine.helpers
+        end
       end
     end
 
-    initializer "action_text.renderer" do |app|
-      app.executor.to_run      { ActionText::Content.renderer = ApplicationController.renderer }
-      app.executor.to_complete { ActionText::Content.renderer = ApplicationController.renderer }
-
+    initializer "action_text.renderer" do
       ActiveSupport.on_load(:action_text_content) do
-        self.renderer = ApplicationController.renderer
+        self.renderer = Class.new(ActionController::Base).renderer
```

Comment:
We can’t use `ActionController::Base.renderer` directly because only children of `ActionController::Base` set up their renderers, right? It’d be convenient for plugins if `ActionController::Base` had a working renderer. I don’t know if there’s a reason it can’t off the top of my head.

Can we safely assume `ActionController::Base` has been loaded here or do we need to wrap this in another load hook?

Reply:
> We can’t use `ActionController::Base.renderer` directly because only children of `ActionController::Base` set up their renderers, right? It’d be convenient for plugins if `ActionController::Base` had a working renderer. I don’t know if there’s a reason it can’t off the top of my head.

Correct!  I can work on `ActionController::Base.renderer` returning a usable renderer if you'd like.  One catch would be that *if* the controller class is anonymous, we should ensure all framework code can handle an anonymous controller class.  For example, [this line](https://github.com/rails/rails/blob/f04819972b75dde87276ece7c1ae9c2e57be12f2/actionview/lib/action_view/renderer/abstract_renderer.rb#L95) should be changed to `prefix&.include?(?/)`.  (That line is bypassed in this PR because of [`prefix_partial_path_with_controller_namespace = false`](https://github.com/rails/rails/pull/40222/files#diff-4737c19868c276c33d997d078ca06162R13).)  Although it might be sufficient to override `Class#name` and return a dummy value instead.

> Can we safely assume `ActionController::Base` has been loaded here or do we need to wrap this in another load hook?

I assumed it was safe because it replaces `ApplicationController`.  But also, wrapping in another load hook causes unfortunate complications.  If `ActionController::Base` has not yet been loaded, such as in a test or a mailer, then `ActionText::Content.renderer` will not be set and `ActionText::Content#to_s` will raise an error.